### ISSUE TITLE
assignment 43

### ISSUE THREAD

**Issue by dipaavhad54-gif** [dipaavhad54-gif/python-assignments#1]
import pandas as pd

from sklearn.preprocessing import LabelEncoder
from sklearn.neighbors import KNeighborsClassifier
from sklearn.metrics import accuracy_score
from sklearn.model_selection import train_test_split

# =========================================================
# STEP 1 : GET DATA
# =========================================================

print("=" * 50)
print("STEP 1 : GET DATA")
print("=" * 50)

data = pd.read_csv("MarvellousInfosystems_PlayPredictor.csv")

print(data)

# =========================================================
# STEP 2 : CLEAN, PREPARE AND MANIPULATE DATA
# =========================================================

print("\n")
print("=" * 50)
print("STEP 2 : CLEAN, PREPARE AND MANIPULATE DATA")
print("=" * 50)

# Remove first column
data = data.drop("Unnamed: 0", axis=1)

# Create LabelEncoder
le_weather = LabelEncoder()
le_temperature = LabelEncoder()
le_play = LabelEncoder()

# Convert string into numbers
data["Wether"] = le_weather.fit_transform(data["Wether"])
data["Temperature"] = le_temperature.fit_transform(data["Temperature"])
data["Play"] = le_play.fit_transform(data["Play"])

print("\nAfter Encoding:")
print(data)

# =========================================================
# STEP 3 : TRAIN DATA
# =========================================================

print("\n")
print("=" * 50)
print("STEP 3 : TRAIN DATA")
print("=" * 50)

# X = input/features
X = data[["Wether", "Temperature"]]

# Y = output
Y = data["Play"]

# Create KNN
knn = KNeighborsClassifier(n_neighbors=3)

# Train using complete dataset
knn.fit(X, Y)

print("KNN training completed.")

# =========================================================
# STEP 4 : TEST DATA
# =========================================================

print("\n")
print("=" * 50)
print("STEP 4 : TEST DATA")
print("=" * 50)

print("\nWeather options:")
print("Sunny")
print("Overcast")
print("Rainy")

print("\nTemperature options:")
print("Hot")
print("Mild")
print("Cool")

weather = input("\nEnter Weather: ")
temperature = input("Enter Temperature: ")

# Convert user input into numbers
weather_value = le_weather.transform([weather])[0]
temperature_value = le_temperature.transform([temperature])[0]

# Predict
result = knn.predict([[weather_value, temperature_value]])

# Convert number back to Yes/No
answer = le_play.inverse_transform(result)

print("\nPredicted Result:", answer[0])

# =========================================================
# STEP 5 : CALCULATE ACCURACY
# =========================================================

print("\n")
print("=" * 50)
print("STEP 5 : CALCULATE ACCURACY")
print("=" * 50)

# Divide data into training and testing
X_train, X_test, Y_train, Y_test = train_test_split(
    X,
    Y,
    test_size=0.5,
    random_state=42
)

# Try different values of K
for k in [1, 3, 5]:

    knn = KNeighborsClassifier(n_neighbors=k)

    knn.fit(X_train, Y_train)

    prediction = knn.predict(X_test)

    accuracy = accuracy_score(Y_test, prediction)

    print("K =", k, "Accuracy =", accuracy * 100, "%")
